preloader.go

```diff
@@ -55,6 +56,14 @@ func (p *PreparedMsg) Encode(s Stream, msg interface{}) error {
 		return err
 	}
 	p.encodedData = data
+	if len(data) >= bufferReuseThreshold {
+		if bcodec, ok := rpcInfo.preloaderInfo.codec.(bufferedBaseCodec); ok {
+			p.returnBuffer = func() {
```

Comment:
Yes. I don't use a preloader, but I imagine it exists for a reason. I traced the history behind this and it looks like it was created for applications that want encoding to be separate from transport-related operations. In that case, it will be invoked when the application calls `prepareMsg`, which uses `p.returnBuffer` if the message is a `*PreparedMsg`.

Reply:
I don't see where `PreparedMsg.returnBuffer` is ever called, though(?).

Reply:
The `returnBuffer` value returned in `prepareMsg` is used by `SendMsg`. If the user passes a `*PreparedMsg` to `SendMsg`, this function would eventually be called by the `Done()` call in loopy.